src/Symfony/Component/Decorator/Attribute/Decorate.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Decorator\Attribute;
+
+/**
+ * An attribute that specifies under which decorator a function or method should be wrapped.
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * @experimental
+ */
+#[\Attribute(\Attribute::TARGET_FUNCTION | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
```

Comment:
What about accepting classes also? I see two possibilities:
- Only support callable classes (with `__invoke()` their **only** public method)
- Plug on every method.

We could imagine a repository decorated with some stopwatches to spy on execution times of each query builders. Having the possibility to define the decorator on the class would be nice instead of repeating everything (and we know Doctrine repo are not the smallest classes...).

Reply:
Good idea! I'm keeping it simple for this proposal, but there are a lot of feature around it. Food for future PRs 👍